[2단계미션] 윤재홍미션 제출합니다.

yoonjaehong26 opened the pull request:
정수영 리뷰어님 2주차 리뷰로 다시 만나게 되었습니다!
아직 고칠 부분이 많고 공부해야 할 부분이 많아 남은 리뷰기간동안 더 공부해 고쳐나가 보겠습니다...
***
먼저  과제의 요건충족을 위하여 리스트 컴포넌트를 프록스를 통한 호출을 가능하게 하기 위하여 먼저 리스트의 틀을 구성하는 RestaurantList컴포넌트와 각 리스트의 내용물에 해당하는 RestaurantList컴포넌트로 나누었습니다.

이후 하드코딩되었던 음식점 리스트를 restaurantsData에서 데이터를 App.jsx, RestaurantList ,RestaurantItem의 순으로 단방향 데이터 흐름을 만들어 프록스를 통한 하나의 동적인 컴포넌트로 재구성하였습니다.

이후 filter기능을 만들기 위하여 usestate의 filteredRestaurants인자를 통하여 필터내용을 구성하였고 setfilteredRestaurants인자를 필터컴포넌트에 주어  filteredRestaurants데이터를 다루게 하여 필터기능을 만들었습니다.
***

1주차에 말씀드린 대로 분리할 수 있는 컴포넌트를 분리하려고 고민하였고 pr기준 버튼과 필터 컴포넌트를 분리해 보았습니다.
다만 헤더의 제목과 리스트컴포넌트는 ui로 만들지 않았는데 실제로 프록스를 통한 기능을 완성해 보니 
순수한 형태의 ui 컴포넌트로서 사용이 힘들다고 보았기 때문입니다.
***

과제 도중  맥북과 윈도우를 오가며 작성하다가 eslint,npm오류로 인하여 시간을 많이 쓰게 되어 아직 너무 부족하게 제출하게 되어 죄송합니다 남은 주말과 티키타가기간내에 열심히 고쳐나가겠습니다!

<img width="740" height="473" alt="image" src="https://github.com/user-attachments/assets/fe1aa1aa-f328-4fda-a856-e4da9a9aaa0d" />

https://github.com/user-attachments/assets/894d85d2-ac75-4720-ae55-2a67e64b21d0

Commit:
refactor(/src):index.html dom요소 리액트 컴포턴트로 재구성

css는 전역 App.css로 임시적용함.이미지 파일 연동 안됨

Commit:
refactor(HomeHeader.jsx,RestrauantList.jsx):이미지 파일 컴포넌트에 연동

Commit:
refactor(/src):style.css파일 module.css로 분리,문법에 맞게 수정

Commit:
refactor(/src,eslintrc.cjs,package-lock.json,package.json):eslint설치 밎 적용

1.문자열 큰따움표샤용 => 작은따움표 사용
2.줄바꿈 에어비앤비 컨벤션 적용
3.dom태그 문법 에어비앤비 컨벤션 적용

Commit:
refactor(/src):코드마지막줄 개행 추가

Commit:
fix(App.jsx, App.css):App.module.css와 중복되어 적용되는 css파일 삭제

Commit:
fix(HomeHeader.jsx):HomeHeader.module.css가 적용되지 않은 부분 수정

Commit:
refactor(eslintrc.cjs):코드마지막줄 개행 추가

Commit:
fix(App.jsx, RestaurantList.jsx):파일명, 컴포넌트명이름 불일치 수정

Commit:
feat(/Main):RestaurantList 리스트 목록 RestaurantItem으로 분리

Commit:
fix(eslintrc.cjs, package-lock.json, package.json):eslint버전 안정화

eslint오류로 임시로 eslint버전을 낮게하였음
'linebreak-style': 0: 맥북과 줄바꿈기호 호환성이 맞지 않아 임시 경고무시처리
"react/react-in-jsx-scope": "off",:버전이 낮아 지금은 필요없지만 해당버전에서는 아직 수정 전

Commit:
feat(public/images,App.jsx,/main):usestate로 음식점 리스트 관리기능 추가

이미지 public파일로 이동,음식점 데이터 usestate로 저장하여 동적인 음식점리스트 기능 생성

Commit:
feat(App.jsx,RestaurantCategoryFilter.jsx):필터컴포넌트를 통한 usestate관리 기능 추가

Commit:
refactor(UI):버튼, 필터 컴포넌트 분리하여 ui폴더에 할당

Commit:
refactor():컴포넌트 파일구조 최적화

1.컴포넌트를 ui기능단위로 폴더를 만들어 폴더구조 개선

2.에어비앤비 컨벤션에 따라 각 폴더 내 대표적인 컴포넌트 파일 명 index.jsx로 변경

3.에어비앤비 컨벤션에 따라 컴포넌트 명, 파일 네이밍 준수

Commit:
refactor(App.jsx,AddRestaurantModal.jsx):에어비앤비 컨벤션에 따른 label,button관련 규칙 준수

1. label로 컨트롤할 input 요소를 감싸게 하여 규칙준수
2.button태그에 타입명시

Commit:
refactor(.eslintrc.cjs,AddRestaurantModal.jsx):label태그 관련 eslint 규칙 수정

 <label> 태그가 HtmlFor 값과 input id 값을 연결해서 control을 연결하는 방식과
for 속성 또는 htmlFor 속 컨트롤할 input 요소를 감싸는 두 방식 중 첫번째 방식을 준수하지만
두번째방식은 태그 구조가 달라져 예외처리 하기로 함

Commit:
refactor():modal 틀을 구성하는 컴포넌트 분리및 module.css 재할당

suyoungj commented on .eslintrc.cjs:
```diff
@@ -3,17 +3,22 @@ module.exports = {
-    "plugin:react/recommended",
-    "plugin:react/jsx-runtime",
-    "plugin:react-hooks/recommended",
+    "airbnb",
```
적용하시게 된 계기가 궁금합니다. eslint 에러로 고생하신건 아마 이 부분 때문일 것 같네요 😅

yoonjaehong26 replied:
처음에 eslint 에어비앤비컨벤션을 적용하여 원래 제 스타일과 에어비앤비스타일의 차이를 비교하며 
어떠한 이유로 규칙들이 생기게 되었는 지를 생각하면서 코드를 짜려고 하였습니다.
***

그 과정속에서 맥북과 윈도우를 오가며 작업하다가 npm run dev과 eslint가 작동하지 않고
운영체제와 관련된 줄바꿈 오류가  나서 운영체제 차이의 문제인줄 알아 해매게 되었는데 
eslint airbnb 컨벤션을 읽어 적용하고  npm이 작동되지 않는 문제를 해결하여 
작업을 이어나갈 수 있었습니다.
***

이후 작업환경을 맥북으로 통일하여 문제가 일어나지 않게 관리하고 있고 프로젝트 초기설정과 관련하여
이러한 문제를 docker를 통하여 해결할 수 있다고 하여 적용해 보고자 알아가고 있습니다!

suyoungj commented on src/App.jsx:
```diff
@@ -1,7 +1,34 @@
 function App() {
-  return <h1>Self-Paced React</h1>;
-}
+  const [filteredRestaurants, setfilteredRestaurants] = useState(restaurant);
```
gemini 리뷰처럼 이렇게 camelCase로 컨벤션이 정해진 경우, 읽는 사람이나 추후 사용하는 사람 모두가 덜 헷갈리도록 따르는 것을 권장드려요.
```jsx
const [filteredRestaurants, setFilteredRestaurants] = useState(restaurant);
```

yoonjaehong26 replied:
네 프록스 및 컴포넌트명 관련 네이밍 검토하고 수정 완료하였습니다.
감사합니다! 
77daecdbb5ed8470aa56740334a2b2551c676a3a

suyoungj commented on src/components/Main/RestaurantList/index.jsx:
```diff
@@ -0,0 +1,16 @@
+    <section className={styles.restaurantListContainer}>
+      <ul className={styles.restaurantList}>
+        {restaurants.map((restaurant) => (
+          <RestaurantItem key={restaurant.id} restaurant={restaurant} />
```
React에서 리스트를 렌더링할 때 `key` prop은 왜 필요할까요??
`key` prop을 전달하지 않으면 React는 내부적으로 어떤 값을 `key`로 사용할까요?

yoonjaehong26 replied:
리스트에서 .map문법을 사용할 때와 같이 동일한 컴포넌트를 여러개 사용할 때 key 값을 지정해야한다는것 까지는 알고  있었는데
알아보니 key값을 지정해 주지 않으면 알아서 index를 쓰다고 합니다.
***
다만 고유한 값을 가지는 key와 달리 index는 동적으로 변하기 때문에 컴포넌트의 순서가 달라지기 때문에 
컴포넌트 추적이 어렵다고 하네요😅
컴포넌트가 동적으로 추가되고 변경되는 리액트 환경에서 변경된 요소만 최소한으로 업데이트를 해야 하기 때문에
고유한 key값을 이용하여 비슷한 컴포넌트를 구별하게 됩니다.
***
따라서 key값은 보통 데이터의 id값을 쓰면 되는데 인덱스와 비슷한 이유로 추적이 불가능한 랜덤변수를 쓰면 안되고
중복되는 값을 피하여 고유한 값을 사용하면 된다고 합니다
따라서 리액트사용을 염두한다면 데이터 구성단계에서부터 고유 key값을 미리 생각하는 것이 중요한 것 같습니다!

suyoungj commented on src/components/Main/RestaurantCategoryFilter/index.jsx:
```diff
@@ -0,0 +1,31 @@
+    if (selectedCategory === '전체') {
+      setfilteredRestaurants(restaurants);
+    } else {
+      const SelectedRestaurants = restaurants.filter(
```
React의 경우 PascalCase는 컴포넌트 네이밍에 사용하고, 변수는 camelCase를 사용하는게 일반적이에요.
위의 경우 `selectedRestaurants`를 사용하는게 적절합니다.

yoonjaehong26 replied:
네 리뷰 반영하여 코드수정하였습니다 감사합니다!
9a61ecfd4f2cba745978bb67c872f64f8cafc265

suyoungj commented on src/components/Main/RestaurantCategoryFilter/index.jsx:
```diff
@@ -0,0 +1,31 @@
+import Filter from '../UI/filter';
+
+function RestaurantCategoryFilter({ restaurants, setfilteredRestaurants }) {
+  const CATEGORIES = ['전체', '한식', '중식', '일식', '양식', '아시안', '기타'];
```
이처럼 변하지 않는 상수의 경우 한 번만 생성하면 되므로, 
컴포넌트 외부에 선언하거나 별도의 파일로 분리해서 리렌더링 간에 불필요하게 재생성되지 않도록 하는 것이 좋아요.

```jsx
const CATEGORIES = ['전체', '한식', '중식', '일식', '양식', '아시안', '기타'];

function RestaurantCategoryFilter({ restaurants, setfilteredRestaurants }) {
  return (
    <section className={styles.restaurantFilterContainer}>
      <Filter
        label="음식점 카테고리 필터"
        options={CATEGORIES}
        onChange={handleFilterChange}
      />
    </section>
  );
}
```

`CATEGORIES`는 프로젝트 여러 곳에서 사용될 수 있어 보이는데, 별도의 파일로 분리하고 `src/components/Aside/AddRestaurantModal.jsx` 등에서 재사용하도록 개선할 수 있지 않을까요?

yoonjaehong26 replied:
```js
//restaurantsData.js
const restaurants = [
  {
    id: 'a01',
    name: '피양콩할마니',
    description:
      '평양 출신의 할머니가 수십 년간 운영해온 비지 전문점 피양콩 할마니. 두부를 빼지 않은 되비지를 맛볼 수 있는 곳으로, ‘피양’은 평안도 사투리로 ‘평양’을 의미한다. 딸과 함께 운영하는 이곳에선 맷돌로 직접 간 콩만을 사용하며, 일체의 조미료를 넣지 않은 건강식을 선보인다. 콩비지와 피양 만두가 이곳의 대표 메뉴지만, 할머니가 옛날 방식을 고수하며 만들어내는 비지전골 또한 이 집의 역사를 느낄 수 있는 특별한 메뉴다. 반찬은 손님들이 먹고 싶은 만큼 덜어 먹을 수 있게 준비돼 있다.',
    category: '한식',
    image: '/images/category-korean.png',
  },
```
네  ['전체', '한식', '중식', '일식', '양식', '아시안', '기타']의 배열을 `restaurantsData.js`에 저장하여 import해오는 방식을 적용해 보려 했는데 

` category: '한식',`과 같이 `restaurants`배열 내에 이미 카테고리 정보가 있어 
`const categories = restaurants.map(restaurant => restaurant.category);`와 같이 가져온 다음에 
중복을 제거하는 방식을 떠올려 봤습니다.

두가지 방식을 비교했을 때 첫번째 방식은 카테고리를 고정적으로 사용하여 예외가 발생하지 않지만 다른 카테고리 추가 시 필터컴포넌트를 수정해야 하고
두번째 방식은 `restaurantsData.js`에 따라 자동으로 카테고리가 설정되지만 NULL이나 문자열이 아닌 경우를 고려해야 한다는 단점이 있는 것 같습니다.

두 방식 모두 웹의 방향성에 따라 달라지겠지만 정수영 리뷰어님께서는 이에 대해 제가 놓친 점이나 어떻게 생각하시는 지 여쭤보고 싶습니다.

suyoungj replied:
A1) 카테고리 추가 시 필터 컴포넌트에 수정이 가해져야 한다면, 필터 컴포넌트가 확장에 닫혀있다는 의미이니 확장에 열려있도록 필터 컴포넌트를 수정하는 것이 바람직해 보여요.

A2) 두 번째 방식에서 restaurantsData는 서버에서 불러온 데이터를 모킹한 것으로 보이는데, 실제 서버에서 데이터를 불러올 경우에는 [Nullish](https://developer.mozilla.org/en-US/docs/Glossary/Nullish)나 타입 문제뿐만 아니라 API 요청 자체가 실패할 가능성도 고려해야 합니다.


1. 클라이언트와 서버가 각각 음식점 카테고리 정책을 관리하는 방식
2. 서버에서 정책을 관리하고 클라이언트는 이를 동기화하는 방식

이렇게 볼 수 있겠네요.

두 방식 중 정답은 없습니다.
실무에서는 정책이 매우 복잡한 경우(예: 수백 개의 카테고리)에는 API 요청 실패 가능성이 있더라도 2번 방식을 선택하고, 지금처럼 단순한 케이스에서는 1번 방식을 주로 활용합니다.

suyoungj commented on src/components/UI/button.jsx:
```diff
@@ -0,0 +1,11 @@
+import styles from './button.module.css';
+
+function button() {
```
gemini 리뷰처럼, React 컴포넌트는 `Button`(PascalCase)를 사용하는게 적절합니다.

yoonjaehong26 replied:
네 리뷰 반영하여 코드수정 완료하였습니다 감사합니다!
2116223e02393900e579c621cfe7ba212bcab13e

suyoungj commented on src/components/UI/Filter/index.jsx:
```diff
@@ -0,0 +1,19 @@
+import styles from './filter.module.css';
+
+function Filter({ label, options, onChange }) {
```
좋은 시도네요 👍
프로젝트 규모가 커져서 `Button, Filter`가 재사용될 여지가 많아질수록 유의미한 분리가 될 것 같아요.

yoonjaehong26 replied:
<img width="828" height="519" alt="image" src="https://github.com/user-attachments/assets/780e7fdf-3c77-4d5c-bdd6-4e337193350d" />
모달까지 추가로 분리해 보았고  파일 구조 좀 더 수정해 보았습니다.  감사합니다!

suyoungj commented on src/components/Main/RestaurantCategoryFilter/index.jsx:
```diff
@@ -0,0 +1,31 @@
+      <Filter
+        label="음식점 카테고리 필터"
+        options={CATEGORIES}
+        onChange={handleFilterChange}
```
👍

yoonjaehong26 replied:
감사합니다!

suyoungj commented on src/App.jsx:
```diff
@@ -1,7 +1,34 @@
+import RestaurantList from './components/Main/RestaurantList';
+import RestaurantDetailModal from './components/Aside/RestaurantDetailModal';
+import AddRestaurantModal from './components/Aside/AddRestaurantModal';
+import restaurant from './data/restaurantsData';
```
**레스토랑 목록** 데이터이니 `restaurants` 로 import 하는게 덜 헷갈리지 않을까요?

yoonjaehong26 replied:
목록데이터임을 고려하여 `restaurants`으로 작명하여 `./data/restaurantsData';`와 같이 파일이름도 `restaurants`으로 작명하였었는데 데이터를 가져오는 과정에서 통일성을 지키지 못했던 것 같습니다.
리뷰 반영하여 코드 수정 완료하였습니다!
4ebf9fa77f0c7b738776e99fab33d03a92b5da94

suyoungj commented on src/App.jsx:
```diff
@@ -1,7 +1,34 @@
+      <main>
+        <RestaurantCategoryFilter
+          restaurants={restaurant}
+          setfilteredRestaurants={setfilteredRestaurants}
```
`RestaurantCategoryFilter`는 음식점 목록을 카테고리 별로 필터링하는 역할을 담당하는 컴포넌트로 생각되네요.
그렇다면 모든 음식점 목록 데이터에 대해 알고 있어야 할까요?
카테고리에 대한 데이터와, 카테고리 변경 로직에만 관심을 갖도록 개선해볼 수 있지 않을까요?

yoonjaehong26 replied:
```js
//App.jsx
function App() {
  const [selectedCategory, setSelectedCategory] = useState('전체');

  const filteredRestaurants = selectedCategory === '전체'
    ? restaurants
    : restaurants.filter((e) => e.category === selectedCategory);


  <main>
        <RestaurantCategoryFilter
          setSelectedCategory={setSelectedCategory}
        />
        <RestaurantList restaurants={filteredRestaurants} />
      </main>
```

```js
///RestaurantCategoryFilter/index.jsx
import { CATEGORIES } from '../../../data/restaurantsData';

function RestaurantCategoryFilter({ setSelectedCategory }) {
  const handleCategoryChange = (e) => {
    setSelectedCategory(e.target.value);
  };
  return (
    <section className={styles.restaurantFilterContainer}>
      <Filter
        label="음식점 카테고리 필터"
        options={CATEGORIES}
        onChange={handleCategoryChange}
      />
    </section>

  );
}

```
App.jsx에서 카테고리에 대한 새로운 usestate를 만들지만 Filter컴포넌트에 전달하지 않고 Filter컴포넌트에서도 필터기능을 처리하지 않으며 `setSelectedCategory`프록스만 전달하여 단방향 정보전달과 끌어올리기를 통한 정보전달을 하여 
Filter컴포넌트는 카테고리 변경 로직만을 가지고 App.jsx에서 카테고리 필터기능을 처리하는 방식으로 바꾸어 보았습니다.
***
이 방식을 통하여 필터컴포넌트는 과도한 데이터 접근을 방지하고 카테고리 변경이라는 하나의 기능만을 명확히 가질 수 있게되어
더 깔끔한 구조를 가지게 된 것 같습니다 감사합니다!

Commit:
fix(App.jsx):App.jsx와 HomeHeade.jsx에서 <header>가 중복되던 부분 제거

suyoungj reviewed (changes requested):
재홍님 안녕하세요. 2단계 미션도 잘 구현해 주셨네요 👍

`Button, Filter`와 같이 하나의 역할을 담당하는 컴포넌트로 분리한 시도 좋았어요.
당장은 유의미하게 느껴지지 않을 수 있지만, 시행착오를 겪다 보면 감이 잡히고 재홍님만의 기준이 생기실거라 생각해요.
어떤 목적으로 컴포넌트를 분리했는지, 실제로 그 목적을 달성했는지 체크하고 개선하는 과정을 거치면 도움이 많이 될 거에요.

개선할 여지가 있는 부분과 네이밍 컨벤션 위주로 리뷰 남겼습니다. (gemini도 리뷰를 잘 해주는군요)
리뷰 확인하시면서 이해가 잘 되지 않거나 궁금하신 점은 코멘트 남겨주세요 😄

Commit:
refactor(App.jsx,index.jsx):에어비앤비 컨벤션에 따른 프록스 네이밍 카멜 케이스 적용

Commit:
refactor(RestaurantList.jsx, RestaurantList.module.css):파일명 및 관련 네이밍 오타 수정

Commit:
refactor():에어비앤비 eslint규칙에 따른 코드 수정

1.리액트 17.0버전 이후 jsx확장자 더이상 사용하지 않아서 삭제
2. 새로고침과 같은 의도치 않은 작동을 막기 위하여 버튼 태그에 type="button추가
3. 줄바꿈및 depth길이 통일화
4. 큰따움표("")대신 작은따움표 사용('')
eslintrc 규칙 수정:
1.'linebreak-style': 0: 맥과 윈도우 환경에서 줄바꿈 스타일차이로 인한 임시로 규칙 무시
2."react/react-in-jsx-scope": "off" : 리액트 17.0버전 이후 jsx확장자 더이상 사용하지 않아 규칙 무시

Commit:
refactor(App.jsx):restaurants로 변수 통일성 준수

Commit:
refactor(/RestaurantCategoryFilter):에어비앤비 컨벤션에따른 프록스 네이밍 카멜케이스 적용

Commit:
refactor(App.jsx, App.css,main.jsx): App.module.css가 전역 css인것을 고려하여 .css파일로 변경및 적용

yoonjaehong26 commented on src/App.jsx:
```diff
@@ -1,7 +1,34 @@
+  const [filteredRestaurants, setfilteredRestaurants] = useState(restaurant);
+
+  return (
+    <div className={styles.app}>
```
61071941cfad853c6dc54bc230e9fbff915d4ad9
App.module.css가 전역 css인것을 고려하여 .css파일로 변경및 적용완료하였습니다.

Commit:
fix(HomeHeader.jsx,Button.jsx):module.css 적용 범위 수정

yoonjaehong26 commented on src/components/Header/HomeHeader/HomeHeader.module.css:
```diff
@@ -0,0 +1,22 @@
+}
+
+
+.gnb__button img {
+  display: block;
+  width: 40px;
+  height: 40px;
+  object-fit: contain;
+}
```
8c330cbba3ead5cc9a4db23fa7a003923c7c466f
module.css 적용 범위  리뷰반영하여 수정 완료했습니다.

yoonjaehong26 reviewed (commented):
리뷰 반영하여 코드 수정완료하였습니다! 
코드의 폴더구조를 좀 더 개선하여 기능별로 폴더를 구성하고 indes.jsx 파일명을 통하여 
폴더를 통한 역할분리를 강화해 보았습니다. 추가로 modal틀을 UI컴포넌트로 분리하여
재사용성을 높여 봤습니다.

<img width="2256" height="1126" alt="image" src="https://github.com/user-attachments/assets/5f6b78ea-1c2f-4f23-b7c9-29cba5932a50" />

***
네이밍 컨벤션과 변수명 통일성을 확인하여 수정하였고 css.modul의 범위가 
재대로 적용되지 않은 부분을 수정하였습니다.

***
재미나이 리뷰를 보다가, 2번항목에서,
웹접근성을 위한 스크린 리더나  seo를 위하여` aria-labeldl,alt`를 추가해 보려 
`<Button/ > `컴포넌트가 프록스로` aria-label,alt `값을 받는 방식을 생각해 봤는데
이런 방식이 ui분리를 통한 순수한 틀로써의 초기 목적에 벗어나진 않을 지 고민하고 있습니다.

스크린 리더나  seo를 고려하면 프록스를 활용하는 방식이 맞는 것 같으나
이러한 방식으로 기능을 추가하다 보니 순수한 형태와는 멀어지는 것 같아서
아래와 같은 방식이 맞는 지에 대하여 궁금합니다!
```js
//기존의 Button컴포넌트
function Button() {
  return (
    <button type="button" className={styles.button} aria-label="button">
      <img src="templates\add-button.png" alt="button" />
    </button>
  );
}
```
```js
//ui에 프록스를 통한 기능 추가 상상도
function Button({버튼이미지파일경로, ariaLabel에넣을값, imgAlt에넣을값}) {
  return (
    <button type="button" className={styles.button} aria-label={ariaLabel에넣을값}>
      <img src={버튼이미지파일경로} alt={imgAlt에넣을값} />
    </button>
  );
}
```
<img width="804" height="673" alt="image" src="https://github.com/user-attachments/assets/75854aa5-86f8-478f-b0dc-c00b359dd827" />

Commit:
refactor(Modal):모든 운영체제 호환성을 위하여 이미지 경고 \사용 금지

Commit:
refactor(App.jsx):컴포넌트 import시 시맨틱 태그 중복 문제 해결

yoonjaehong26 commented on src/components/UI/button.jsx:
```diff
@@ -0,0 +1,11 @@
+import styles from './button.module.css';
+
+function button() {
```
```js
function Button() {
  return (
    <button type="button" className={styles.button} aria-label="button">
      <img src="templates\add-button.png" alt="button" />
    </button>
  );
}
```
컴포넌트 네이밍 pascalCase로 적용하여 코드 수정 완료하였습니다.

Commit:
fix(eslintrc.cjs,App.jsx,main.jsx):eslintrc rules 중첩 오류 수정,문자영 자은따움표 사용,빈 프래그먼트 사용 금지

suyoungj reviewed (approved):
꼼꼼히 반영해주셔서 감사합니다 재홍님 👍
질문드린 내용에 대해 깊이 있게 학습해주신 점도 좋았어요.

> 재미나이 리뷰를 보다가, 2번항목에서,
웹접근성을 위한 스크린 리더나 seo를 위하여 aria-labeldl,alt를 추가해 보려
<Button/ > 컴포넌트가 프록스로aria-label,alt값을 받는 방식을 생각해 봤는데
이런 방식이 ui분리를 통한 순수한 틀로써의 초기 목적에 벗어나진 않을 지 고민하고 있습니다.

말씀하신 목적에 벗어나는 경우는 button이 UI와 클릭 이벤트만 처리하는 것이 아니라, 도메인 로직까지 button이 알게 되고, 이를 주입받아 직접 처리하려는 상황과 같은 경우일 거에요. 현재 방향성이라면 크게 목적에 벗어나지 않아 보입니다.
img `src`와 `alt`의 경우 `<img>`의 어트리뷰트이니, `<img>` 자체를 `children` prop으로 처리하면 불필요한 prop을 전달하지 않아도 되겠네요. 또한 `<img>` 뿐만 아니라 다른 요소를 button 내부에 렌더링하기 위해 `children` prop을 활용할 수도 있어요.

React, JS뿐만 아니라 HTML, CSS에서 제공하는 스펙으로도 간단히 문제가 해결되는 경우가 많으니, 리뷰 과정에서 답변주신 것처럼 골고루 학습하시는 것을 추천드려요.
3주차 미션도 고생 많으셨습니다 👏

Commit:
refactor(App.jsx,index.jsx,restaurantsData.js):filter컴포넌트 에서 App.jsx로 카테고리변경기능 이전

suyoungj commented:
### 1. PR 요청한 원본 저장소(cho-log/self-paced-react) upstream remote 추가
```bash
# 본인 브랜치(step2React)에서 upstream remote 추가
git remote add upstream https://github.com/cho-log/self-paced-react.git

# upstream 동기화
git fetch upstream
```

### 2. 소스 제어 -> 분기 -> 병합 클릭
<img width="1622" height="1148" alt="image" src="https://github.com/user-attachments/assets/728dc4af-7191-4562-a25c-faca4adfdfb1" />

### 3. PR 요청한 원본 저장소의 브랜치(`upstream/yoonjaehong26`) 클릭해서 merge 시도 (컨플릭트 발생)
<img width="998" height="518" alt="image" src="https://github.com/user-attachments/assets/5601197d-316c-4fe7-9f15-87e577fafdec" />
<img width="255" height="257" alt="image" src="https://github.com/user-attachments/assets/97f159c2-d1b1-44c2-8f30-6bb815c3dbab" />

- 컨플릭트 해결 후 변경사항 push 해주시면 됩니다.

Commit:
Merge remote-tracking branch 'upstream/yoonjaehong26' into step2React

Commit:
fix():병합과정에서바뀌지않은부분수정

Commit:
fix():병합 시 중복됬던 파일들 삭제
